chia/seeder/dns_server.py

```diff
@@ -69,18 +71,25 @@ class DNSServer:
     lock: asyncio.Lock
     pointer: int
     crawl_db: aiosqlite.Connection
+    prometheus: PrometheusSeeder
 
-    def __init__(self):
+    def __init__(self, config: Dict, root_path: Path):
         self.reliable_peers_v4 = []
         self.reliable_peers_v6 = []
         self.lock = asyncio.Lock()
         self.pointer_v4 = 0
         self.pointer_v6 = 0
-        db_path_replaced: str = "crawler.db"
-        root_path = DEFAULT_ROOT_PATH
-        self.db_path = path_from_root(root_path, db_path_replaced)
+
+        if "crawler_db_path" in config and config["crawler_db_path"] != "":
+            path = Path(config["crawler_db_path"])
+            self.db_path = path.resolve()
```

Comment:
Shouldn't this be resolved relative to the root path as well?

Reply:
Not necessarily. The only place I know this is used (because we added it for this) is in a kubernetes cluster, where we mount a volume to a different location outside of the .chia directory to persist the crawler DB to - so when we set this config option, its a full path

Reply:
Maybe require it be absolute then?  I would expect that config paths getting resolved relative to the working directory would be a fairly rare desire.

Reply:
I changed this to `path.expanduser` to account for a path that might be `~/path/to/something`, but it now otherwise requires the path to be absolute